R/num_equal.R

```diff
@@ -46,3 +45,16 @@ num_equal <- function(x, y, tolerance = default_tol()) {
 
   avg_diff < tolerance
 }
+
+
+can_int64 <- function(x) {
+  if (is.integer(x) || inherits(x, "integer64")) {
+    return(TRUE)
+  }
+
+  # https://yutani.rbind.io/post/savvy-v0.7.1-usize/
+  in_range <- x >= -2^53 & x <= 2^53 - 1
```

Comment:
```suggestion
  in_range <- x >= -2^53 & x <= 2^53
```

2^53 is representable directly